annotations.go

```diff
@@ -0,0 +1,209 @@
+		{
+			Name:      "create",
+			Usage:     "create a graph annotation",
+			ArgsUsage: "[--title <title>] [--description <descriptio>] [--from <from>] [--to <to>] [--service -s <service>] [--role -r <role>]",
+			Description: `
+    Creates a graph annotation.
+`,
+			Action: doAnnotationsCreate,
+			Flags: []cli.Flag{
+				cli.StringFlag{Name: "title", Usage: "Title for annotation"},
+				cli.StringFlag{Name: "description", Usage: "Description for annotation"},
+				cli.IntFlag{Name: "from", Usage: "Starting time (epoch seconds)"},
+				cli.IntFlag{Name: "to", Usage: "Ending time (epoch seconds)"},
+				cli.StringFlag{Name: "service, s", Usage: "Service name for annotation"},
+				cli.StringSliceFlag{
+					Name:  "role, r",
+					Value: &cli.StringSlice{},
+					Usage: "Roles for annotation. Multiple choices are allowed",
+				},
+			},
+		},
+		{
+			Name:      "list",
+			Usage:     "list annotations",
+			ArgsUsage: "[--from <from>] [--to <to>] [--service -s <service>]",
+			Description: `
+    Shows annotations by service name and duration (from and to)
+`,
+			Action: doAnnotationsList,
+			Flags: []cli.Flag{
+				cli.StringFlag{Name: "service, s", Usage: "Service name for annotation"},
+				cli.IntFlag{Name: "from", Usage: "Starting time (epoch seconds)"},
+				cli.IntFlag{Name: "to", Usage: "Ending time (epoch seconds)"},
+			},
+		},
+		{
+			Name:      "update",
+			Usage:     "update annotation",
+			ArgsUsage: "[--id <id>] [--title <title>] [--description <descriptio>] [--from <from>] [--to <to>] [--service -s <service>] [--role -r <role>]",
+			Description: `
+    Updates an annotation
+`,
+			Action: doAnnotationsUpdate,
+			Flags: []cli.Flag{
+				cli.StringFlag{Name: "id", Usage: "Annotation ID."},
+				cli.StringFlag{Name: "service, s", Usage: "Service name for annotation"},
+				cli.StringFlag{Name: "title", Usage: "Title for annotation"},
+				cli.StringFlag{Name: "description", Usage: "Description for annotation"},
+				cli.IntFlag{Name: "from", Usage: "Starting time (epoch seconds)"},
+				cli.IntFlag{Name: "to", Usage: "Ending time (epoch seconds)"},
+				cli.StringSliceFlag{
+					Name:  "role, r",
+					Value: &cli.StringSlice{},
+					Usage: "Roles for annotation. Multiple choices are allowed",
+				},
+			},
+		},
+		{
+			Name:      "delete",
+			Usage:     "delete annotation",
+			ArgsUsage: "[--id <id>]",
+			Description: `
+    Delete graph annotation by annotation id
+`,
+			Action: doAnnotationsDelete,
+			Flags: []cli.Flag{
+				cli.StringFlag{Name: "id", Usage: "Graph annotation ID"},
+			},
+		},
+	},
+}
+
+func doAnnotationsCreate(c *cli.Context) error {
+	title := c.String("title")
+	description := c.String("description")
+	from := c.Int64("from")
+	to := c.Int64("to")
+	service := c.String("service")
+	roles := c.StringSlice("role")
+
+	if service == "" {
+		_ = cli.ShowCommandHelp(c, "create")
+		return cli.NewExitError("`service` is a required field to create a graph annotation.", 1)
+	}
```

Comment:
`mkr annotations create` without `--title` should output usage help IMO.  `mkr annotations create --service ExampleService --from 1486962935 --to 1486962940` prints 400 Bad Request but user cannot tell why.